From a903b134411d7e324ae79eeb8a2394f9466e448e Mon Sep 17 00:00:00 2001
From: Thomas Wu <quic_wthomas@quicinc.com>
Date: Thu, 21 Dec 2023 10:29:50 -0800
Subject: [PATCH] wifi: ath12k: Disable B-TWT feature bit

Disable B-TWT feature in HE MAC element
field. Feature is not supported currently for Access
Point.

Signed-off-by: Thomas Wu <quic_wthomas@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index e6baf16..462176f 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -8574,6 +8574,7 @@ static void ath12k_mac_copy_he_cap(struct ath12k *ar,
 
 	switch (iftype) {
 	case NL80211_IFTYPE_AP:
+		he_cap_elem->mac_cap_info[2] &= ~IEEE80211_HE_MAC_CAP2_BCAST_TWT;
 		he_cap_elem->phy_cap_info[3] &=
 			~IEEE80211_HE_PHY_CAP3_DCM_MAX_CONST_TX_MASK;
 		break;
-- 
2.25.1

